main.go

```diff
@@ -39,36 +42,57 @@ func main() {
 	// - TODO(displague) VolumeAttachments
 	// - TODO(displague) Volumes (are these project specific?)
 
-	devices, _, err := client.Devices.List(projectID, nil)
+	devices, err := getAllProjectDevices(projectID)
 
 	if err != nil {
 		warn.Println("Could not list devices", err)
 	}
 
 	for _, device := range devices {
 		fmt.Println("Deleting device", device.Hostname)
-		if _, err := client.Devices.Delete(device.ID, true); err != nil {
+		if _, err := client.DevicesApi.DeleteDevice(context.Background(), device.GetId()).Execute(); err != nil {
 			warn.Println("Could not delete device", err)
 		}
 	}
 
-	vlans, _, err := client.ProjectVirtualNetworks.List(projectID, nil)
+	vlans, _, err := client.VLANsApi.FindVirtualNetworks(context.Background(), projectID).Execute()
 
 	if err != nil {
 		warn.Println("Could not list vlans", err)
 	}
 
 	for _, vlan := range vlans.VirtualNetworks {
 		fmt.Println("Deleting vlan", vlan.Description)
-		if _, err := client.ProjectVirtualNetworks.Delete(vlan.ID); err != nil {
+		if _, _, err := client.VLANsApi.DeleteVirtualNetwork(context.Background(), vlan.GetId()).Execute(); err != nil {
 			warn.Println("Could not delete vlan", err)
 		}
 	}
 
 	if os.Getenv(keepProjectEnv) == "false" {
-		_, err := client.Projects.Delete(projectID)
+		_, err := client.ProjectsApi.DeleteProject(context.Background(), projectID).Execute()
 		if err != nil {
 			warn.Println("Could not delete project", err)
 		}
 	}
 }
+
+func getAllProjectDevices(projectID string) ([]metal.Device, error) {
+	var devices []metal.Device
```

Comment:
This could be copied in to metal-go's examples to satisfy equinix-labs/metal-go#6